### PR TITLE
Upgrade Hazelcast example with the latest hazelcast-kubernetes-bootst…

### PR DESCRIPTION
**Pull request by oscerd**
…rapper 0.8.0, using Deployments instead of Replication Controller.

**What this PR does / why we need it**:
This PR upgrade the Hazelcast example in storage folder. It would be great to be aligned with the latest version of hazelcast-kubernetes-bootstrapper

### CODE REVIEW

**Comment by oscerd** [comment]
I signed it

**Comment by thockin** [comment]
@pires can you take a look?

**Comment by pires** [comment]
@thockin I will first thing tomorrow.

@oscerd I'm about to release a new version, will ping you here when I release it tomorrow. Thanks for picking this!

File: examples/storage/hazelcast/README.md [hunk 1]
```diff
@@ -65,52 +65,41 @@ $ kubectl create -f examples/storage/hazelcast/hazelcast-service.yaml
 
 The real power of Kubernetes and Hazelcast lies in easily building a replicated, resizable Hazelcast cluster.
 
-In Kubernetes a _[Replication Controller](../../../docs/user-guide/replication-controller.md)_ is responsible for replicating sets of identical pods.  Like a _Service_ it has a selector query which identifies the members of it's set.  Unlike a _Service_ it also has a desired number of replicas, and it will create or delete _Pods_ to ensure that the number of _Pods_ matches up with it's desired state.
+In Kubernetes a _[_Deployment_](../../../docs/user-guide/deployments.md)_ is responsible for replicating sets of identical pods. Like a _Service_ it has a selector query which identifies the members of it's set.  Unlike a _Service_ it also has a desired number of replicas, and it will create or delete _Pods_ to ensure that the number of _Pods_ matches up with it's desired state.
```
Comment (pires) [inline examples/storage/hazelcast/README.md]: "its set"

File: examples/storage/hazelcast/README.md [hunk 2]
```diff
@@ -65,52 +65,41 @@ $ kubectl create -f examples/storage/hazelcast/hazelcast-service.yaml
...
 ```
 
-[Download example](hazelcast-controller.yaml?raw=true)
+[Download example](hazelcast-deployment.yaml?raw=true)
 <!-- END MUNGE: EXAMPLE hazelcast-controller.yaml -->
 
-There are a few things to note in this description.  First is that we are running the `quay.io/pires/hazelcast-kubernetes` image, tag `0.5`.  This is a `busybox` installation with JRE 8 Update 45.  However it also adds a custom [`application`](https://github.com/pires/hazelcast-kubernetes-bootstrapper) that finds any Hazelcast nodes in the cluster and bootstraps an Hazelcast instance accordingly.  The `HazelcastDiscoveryController` discovers the Kubernetes API Server using the built in Kubernetes discovery service, and then uses the Kubernetes API to find new nodes (more on this later).
-
-You may also note that we tell Kubernetes that the container exposes the `hazelcast` port.  Finally, we tell the cluster manager that we need 1 cpu core.
+You may note that we tell Kubernetes that the container exposes the `hazelcast` port. Finally, we tell the cluster manager that we need 1 cpu core.
```
Comment (pires) [inline examples/storage/hazelcast/README.md]: Where is "1 cpu core" requirement/limit defined?

File: examples/storage/hazelcast/image/Dockerfile [hunk 3]
```diff
@@ -12,13 +12,13 @@
...
 
 
 EXPOSE 5701
 
 RUN \
-  curl -Lskj https://github.com/pires/hazelcast-kubernetes-bootstrapper/releases/download/0.5/hazelcast-kubernetes-bootstrapper-0.5.jar \
+  curl -Lskj https://github.com/pires/hazelcast-kubernetes-bootstrapper/releases/download/0.7.0/hazelcast-kubernetes-bootstrapper-0.7.0.jar \
   -o /bootstrapper.jar
 
 CMD java -jar /bootstrapper.jar
```
Comment (pires) [inline examples/storage/hazelcast/image/Dockerfile]: Why is the Dockerfile included if the image used here is from my quay.io repo?

**Review by pires — CHANGES_REQUESTED** [review]

**Comment by oscerd** [comment]
@pires @thockin I'll update soon the PR and wait for the next release

**Comment by oscerd** [comment]
PR updated.

**Comment by pires** [comment]
@oscerd version 0.8.0 released and updated instructions.

**Comment by oscerd** [comment]
I will align to 0.8.0

**Comment by oscerd** [comment]
@pires @thockin it should be fine

**Review by pires — APPROVED** [review]
/lgtm

**Comment by pires** [comment]
@thockin need you or someone to remove the release-note-label-needed label.

**Comment by pires** [comment]
Will the bot ignore `release-note-label-needed`? /cc @apelisse

**Comment by thockin** [comment]
@k8s-bot ok to test

**Comment by pires** [comment]
@oscerd seems you have to add/update a test.

**Comment by oscerd** [comment]
I hope now it's fine

**Comment by oscerd** [comment]
We need just the lgtm label

**Comment by oscerd** [comment]
Any news?

**Comment by oscerd** [comment]
Not sure, but the test error doesn't look something related to this PR

**Comment by pires** [comment]
@k8s-bot kops aws e2e test this

**Comment by oscerd** [comment]
It's OK now

**Comment by yujuhong** [comment]
The Hazelcast tests have been consistently failing since this PR was merged: https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-examples/
Could you please check the failures? @oscerd @pires

**Comment by oscerd** [comment]
The hazelcast e2e test is fixed with this PR: https://github.com/kubernetes/kubernetes/pull/41310

**Pull request merged**